Allow wild card type for gtm wide ip

papineni87 opened the pull request:
Fix for #1920

wojtek0806 reviewed (changes requested):
add a functional test that tests this scenario

wojtek0806 commented on ansible_collections/f5networks/f5_modules/plugins/modules/bigip_gtm_wide_ip.py:
```diff
@@ -387,7 +387,7 @@ class ModuleParameters(Parameters):
     def last_resort_pool(self):
         if self._values['last_resort_pool'] in [None, '', 'none']:
-            return ''
+            return None
```
should be 'none' not None as it won't send anything to the device ? what is the fix here ? Last resort pool API accepts 'none' as parameter to effectively remove last resort pool setting, or am i missing something?

papineni87 replied:
The iControl REST API no longer accepts an empty 'lastResortPool' property for wide IP objects. 

We are using None in the rest of the code, so i left it as None to make consistency and it fixed the issue. I tried "none" and it has same result, if you want change it to "none" instead of None

wojtek0806 replied:
have you checked on earlier versions of bigip you must not break other versions with this

papineni87 replied:
Tested in bigip v13.0, v14.1.3,v15.0. along with v14.1.4

wojtek0806 reviewed (changes requested):
Make the necessary changes and post test run results

papineni87 commented:
Adding test run results :

TASK [bigip_gtm_wide_ip : Issue 01920 - Include issue variables] ************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:3
ok: [bigip1] => {
    "ansible_facts": {
        "pool_lb_method": "round-robin",
        "pool_name": "issue-01920",
        "wide_ip1": "*.apps.example.com"
    },
    "ansible_included_var_files": [
        "/Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/vars/issue-01920.yaml"
    ],
    "changed": false
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Provision GTM] **********************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:7
<10.144.72.9> ESTABLISH LOCAL CONNECTION FOR USER: papineni
<10.144.72.9> EXEC /bin/sh -c 'echo ~papineni && sleep 0'
<10.144.72.9> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/papineni/.ansible/tmp `"&& mkdir "` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076760.5277388-36698-191008929399390 `" && echo ansible-tmp-1619076760.5277388-36698-191008929399390="` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076760.5277388-36698-191008929399390 `" ) && sleep 0'
Using module file /Users/papineni/ansible_repo/f5-ansible/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_provision.py
<10.144.72.9> PUT /Users/papineni/.ansible/tmp/ansible-local-343978uc5_sjj/tmpep31vmdk TO /Users/papineni/.ansible/tmp/ansible-tmp-1619076760.5277388-36698-191008929399390/AnsiballZ_bigip_provision.py
<10.144.72.9> EXEC /bin/sh -c 'chmod u+x /Users/papineni/.ansible/tmp/ansible-tmp-1619076760.5277388-36698-191008929399390/ /Users/papineni/.ansible/tmp/ansible-tmp-1619076760.5277388-36698-191008929399390/AnsiballZ_bigip_provision.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'F5_SERVER=10.144.72.9 F5_USER=admin F5_PASSWORD=F5site02 F5_SERVER_PORT=443 F5_VALIDATE_CERTS=False F5_TEEM=True /Users/papineni/ansible_dev/bin/python /Users/papineni/.ansible/tmp/ansible-tmp-1619076760.5277388-36698-191008929399390/AnsiballZ_bigip_provision.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'rm -f -r /Users/papineni/.ansible/tmp/ansible-tmp-1619076760.5277388-36698-191008929399390/ > /dev/null 2>&1 && sleep 0'
changed: [bigip1] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "level": "nominal",
            "memory": null,
            "module": "gtm",
            "provider": {
                "auth_provider": null,
                "no_f5_teem": null,
                "password": null,
                "server": null,
                "server_port": null,
                "timeout": null,
                "transport": "rest",
                "user": null,
                "validate_certs": null
            },
            "state": "present"
        }
    },
    "level": "nominal"
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Collect BIG-IP facts] ***************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:13
<10.144.72.9> ESTABLISH LOCAL CONNECTION FOR USER: papineni
<10.144.72.9> EXEC /bin/sh -c 'echo ~papineni && sleep 0'
<10.144.72.9> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/papineni/.ansible/tmp `"&& mkdir "` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076812.94894-36720-157006445171114 `" && echo ansible-tmp-1619076812.94894-36720-157006445171114="` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076812.94894-36720-157006445171114 `" ) && sleep 0'
Using module file /Users/papineni/ansible_repo/f5-ansible/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_device_info.py
<10.144.72.9> PUT /Users/papineni/.ansible/tmp/ansible-local-343978uc5_sjj/tmpad1l50np TO /Users/papineni/.ansible/tmp/ansible-tmp-1619076812.94894-36720-157006445171114/AnsiballZ_bigip_device_info.py
<10.144.72.9> EXEC /bin/sh -c 'chmod u+x /Users/papineni/.ansible/tmp/ansible-tmp-1619076812.94894-36720-157006445171114/ /Users/papineni/.ansible/tmp/ansible-tmp-1619076812.94894-36720-157006445171114/AnsiballZ_bigip_device_info.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'F5_SERVER=10.144.72.9 F5_USER=admin F5_PASSWORD=F5site02 F5_SERVER_PORT=443 F5_VALIDATE_CERTS=False F5_TEEM=True /Users/papineni/ansible_dev/bin/python /Users/papineni/.ansible/tmp/ansible-tmp-1619076812.94894-36720-157006445171114/AnsiballZ_bigip_device_info.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'rm -f -r /Users/papineni/.ansible/tmp/ansible-tmp-1619076812.94894-36720-157006445171114/ > /dev/null 2>&1 && sleep 0'
ok: [bigip1] => {
    "ansible_facts": {
        "ansible_net_queried": true,
        "ansible_net_system_info": {
            "base_mac_address": "fa:16:3e:40:ca:5a",
            "chassis_serial": "42250b55-61ff-9357-c8a22e23f83a",
            "hardware_information": [
                {
                    "model": "Virtual disk",
                    "name": "HD1",
                    "type": "physical-disk",
                    "versions": [
                        {
                            "name": "SerialNumber",
                            "version": "VMware-sda"
                        },
                        {
                            "name": "Size",
                            "version": "160.00G"
                        },
                        {
                            "name": "Firmware Version",
                            "version": "2.0"
                        },
                        {
                            "name": "Media Type",
                            "version": "HDD"
                        }
                    ]
                },
                {
                    "model": "Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz",
                    "name": "cpus",
                    "type": "base-board",
                    "versions": [
                        {
                            "name": "cache size",
                            "version": "35840 KB"
                        },
                        {
                            "name": "cores",
                            "version": "8  (physical:8)"
                        },
                        {
                            "name": "cpu MHz",
                            "version": "2593.993"
                        },
                        {
                            "name": "cpu sockets",
                            "version": "8"
                        },
                        {
                            "name": "cpu stepping",
                            "version": "0"
                        }
                    ]
                }
            ],
            "marketing_name": "BIG-IP Virtual Edition",
            "package_edition": "Final",
            "package_version": "Build 0.0.11 - Thu Feb 11 19:05:03 PST 2021",
            "platform": "Z100",
            "product_build": "0.0.11",
            "product_build_date": "Thu Feb 11 19:05:03 PST 2021",
            "product_built": 210211190503,
            "product_changelist": 3455333,
            "product_code": "BIG-IP",
            "product_jobid": 1274147,
            "product_version": "14.1.4",
            "time": {
                "day": 22,
                "hour": 7,
                "minute": 33,
                "month": 4,
                "second": 40,
                "year": 2021
            },
            "uptime": 256456
        }
    },
    "changed": false,
    "invocation": {
        "module_args": {
            "gather_subset": [
                "system-info"
            ],
            "include": "system-info",
            "provider": {
                "auth_provider": null,
                "no_f5_teem": null,
                "password": null,
                "server": null,
                "server_port": null,
                "timeout": null,
                "transport": "rest",
                "user": null,
                "validate_certs": null
            }
        }
    },
    "queried": true,
    "system_info": {
        "base_mac_address": "fa:16:3e:40:ca:5a",
        "chassis_serial": "42250b55-61ff-9357-c8a22e23f83a",
        "hardware_information": [
            {
                "model": "Virtual disk",
                "name": "HD1",
                "type": "physical-disk",
                "versions": [
                    {
                        "name": "SerialNumber",
                        "version": "VMware-sda"
                    },
                    {
                        "name": "Size",
                        "version": "160.00G"
                    },
                    {
                        "name": "Firmware Version",
                        "version": "2.0"
                    },
                    {
                        "name": "Media Type",
                        "version": "HDD"
                    }
                ]
            },
            {
                "model": "Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz",
                "name": "cpus",
                "type": "base-board",
                "versions": [
                    {
                        "name": "cache size",
                        "version": "35840 KB"
                    },
                    {
                        "name": "cores",
                        "version": "8  (physical:8)"
                    },
                    {
                        "name": "cpu MHz",
                        "version": "2593.993"
                    },
                    {
                        "name": "cpu sockets",
                        "version": "8"
                    },
                    {
                        "name": "cpu stepping",
                        "version": "0"
                    }
                ]
            }
        ],
        "marketing_name": "BIG-IP Virtual Edition",
        "package_edition": "Final",
        "package_version": "Build 0.0.11 - Thu Feb 11 19:05:03 PST 2021",
        "platform": "Z100",
        "product_build": "0.0.11",
        "product_build_date": "Thu Feb 11 19:05:03 PST 2021",
        "product_built": 210211190503,
        "product_changelist": 3455333,
        "product_code": "BIG-IP",
        "product_jobid": 1274147,
        "product_version": "14.1.4",
        "time": {
            "day": 22,
            "hour": 7,
            "minute": 33,
            "month": 4,
            "second": 40,
            "year": 2021
        },
        "uptime": 256456
    }
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Create pool, type 'a'] **************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:18
<10.144.72.9> ESTABLISH LOCAL CONNECTION FOR USER: papineni
<10.144.72.9> EXEC /bin/sh -c 'echo ~papineni && sleep 0'
<10.144.72.9> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/papineni/.ansible/tmp `"&& mkdir "` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076828.8811252-36740-180006779928013 `" && echo ansible-tmp-1619076828.8811252-36740-180006779928013="` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076828.8811252-36740-180006779928013 `" ) && sleep 0'
Using module file /Users/papineni/ansible_repo/f5-ansible/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_gtm_pool.py
<10.144.72.9> PUT /Users/papineni/.ansible/tmp/ansible-local-343978uc5_sjj/tmppjk4zfir TO /Users/papineni/.ansible/tmp/ansible-tmp-1619076828.8811252-36740-180006779928013/AnsiballZ_bigip_gtm_pool.py
<10.144.72.9> EXEC /bin/sh -c 'chmod u+x /Users/papineni/.ansible/tmp/ansible-tmp-1619076828.8811252-36740-180006779928013/ /Users/papineni/.ansible/tmp/ansible-tmp-1619076828.8811252-36740-180006779928013/AnsiballZ_bigip_gtm_pool.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'F5_SERVER=10.144.72.9 F5_USER=admin F5_PASSWORD=F5site02 F5_SERVER_PORT=443 F5_VALIDATE_CERTS=False F5_TEEM=True /Users/papineni/ansible_dev/bin/python /Users/papineni/.ansible/tmp/ansible-tmp-1619076828.8811252-36740-180006779928013/AnsiballZ_bigip_gtm_pool.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'rm -f -r /Users/papineni/.ansible/tmp/ansible-tmp-1619076828.8811252-36740-180006779928013/ > /dev/null 2>&1 && sleep 0'
changed: [bigip1] => {
    "changed": true,
    "enabled": true,
    "invocation": {
        "module_args": {
            "alternate_lb_method": null,
            "availability_requirements": null,
            "fallback_ip": null,
            "fallback_lb_method": null,
            "max_answers_returned": null,
            "members": null,
            "monitors": null,
            "name": "issue-01920",
            "partition": "Common",
            "preferred_lb_method": "round-robin",
            "provider": {
                "auth_provider": null,
                "no_f5_teem": null,
                "password": null,
                "server": null,
                "server_port": null,
                "timeout": null,
                "transport": "rest",
                "user": null,
                "validate_certs": null
            },
            "state": "present",
            "ttl": null,
            "type": "a"
        }
    },
    "monitors": [],
    "preferred_lb_method": "round-robin"
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Create wide IP] *********************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:27
<10.144.72.9> ESTABLISH LOCAL CONNECTION FOR USER: papineni
<10.144.72.9> EXEC /bin/sh -c 'echo ~papineni && sleep 0'
<10.144.72.9> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/papineni/.ansible/tmp `"&& mkdir "` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076839.630785-36760-52409937959547 `" && echo ansible-tmp-1619076839.630785-36760-52409937959547="` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076839.630785-36760-52409937959547 `" ) && sleep 0'
Using module file /Users/papineni/ansible_repo/f5-ansible/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_gtm_wide_ip.py
<10.144.72.9> PUT /Users/papineni/.ansible/tmp/ansible-local-343978uc5_sjj/tmpd4ot9m6_ TO /Users/papineni/.ansible/tmp/ansible-tmp-1619076839.630785-36760-52409937959547/AnsiballZ_bigip_gtm_wide_ip.py
<10.144.72.9> EXEC /bin/sh -c 'chmod u+x /Users/papineni/.ansible/tmp/ansible-tmp-1619076839.630785-36760-52409937959547/ /Users/papineni/.ansible/tmp/ansible-tmp-1619076839.630785-36760-52409937959547/AnsiballZ_bigip_gtm_wide_ip.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'F5_SERVER=10.144.72.9 F5_USER=admin F5_PASSWORD=F5site02 F5_SERVER_PORT=443 F5_VALIDATE_CERTS=False F5_TEEM=True /Users/papineni/ansible_dev/bin/python /Users/papineni/.ansible/tmp/ansible-tmp-1619076839.630785-36760-52409937959547/AnsiballZ_bigip_gtm_wide_ip.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'rm -f -r /Users/papineni/.ansible/tmp/ansible-tmp-1619076839.630785-36760-52409937959547/ > /dev/null 2>&1 && sleep 0'
changed: [bigip1] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "aliases": null,
            "irules": null,
            "last_resort_pool": null,
            "lb_method": "round-robin",
            "name": "*.apps.example.com",
            "partition": "Common",
            "persist_cidr_ipv4": null,
            "persist_cidr_ipv6": null,
            "persistence": null,
            "persistence_ttl": null,
            "pool_lb_method": "round-robin",
            "pools": [
                {
                    "name": "issue-01920",
                    "order": null,
                    "ratio": null
                }
            ],
            "provider": {
                "auth_provider": null,
                "no_f5_teem": null,
                "password": null,
                "server": null,
                "server_port": null,
                "timeout": null,
                "transport": "rest",
                "user": null,
                "validate_certs": null
            },
            "state": "present",
            "type": "a"
        }
    },
    "lb_method": "round-robin",
    "name": "*.apps.example.com",
    "pool_lb_method": "round-robin",
    "pools": [
        {
            "name": "/Common/issue-01920",
            "order": 0,
            "ratio": 0
        }
    ],
    "state": "present"
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Assert Create wide IP] **************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:38
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Create wide IP - Idempotent check] **************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:44
<10.144.72.9> ESTABLISH LOCAL CONNECTION FOR USER: papineni
<10.144.72.9> EXEC /bin/sh -c 'echo ~papineni && sleep 0'
<10.144.72.9> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/papineni/.ansible/tmp `"&& mkdir "` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076849.0194519-36782-130389179744649 `" && echo ansible-tmp-1619076849.0194519-36782-130389179744649="` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076849.0194519-36782-130389179744649 `" ) && sleep 0'
Using module file /Users/papineni/ansible_repo/f5-ansible/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_gtm_wide_ip.py
<10.144.72.9> PUT /Users/papineni/.ansible/tmp/ansible-local-343978uc5_sjj/tmp8qaot42n TO /Users/papineni/.ansible/tmp/ansible-tmp-1619076849.0194519-36782-130389179744649/AnsiballZ_bigip_gtm_wide_ip.py
<10.144.72.9> EXEC /bin/sh -c 'chmod u+x /Users/papineni/.ansible/tmp/ansible-tmp-1619076849.0194519-36782-130389179744649/ /Users/papineni/.ansible/tmp/ansible-tmp-1619076849.0194519-36782-130389179744649/AnsiballZ_bigip_gtm_wide_ip.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'F5_SERVER=10.144.72.9 F5_USER=admin F5_PASSWORD=F5site02 F5_SERVER_PORT=443 F5_VALIDATE_CERTS=False F5_TEEM=True /Users/papineni/ansible_dev/bin/python /Users/papineni/.ansible/tmp/ansible-tmp-1619076849.0194519-36782-130389179744649/AnsiballZ_bigip_gtm_wide_ip.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'rm -f -r /Users/papineni/.ansible/tmp/ansible-tmp-1619076849.0194519-36782-130389179744649/ > /dev/null 2>&1 && sleep 0'
ok: [bigip1] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "aliases": null,
            "irules": null,
            "last_resort_pool": null,
            "lb_method": "round-robin",
            "name": "*.apps.example.com",
            "partition": "Common",
            "persist_cidr_ipv4": null,
            "persist_cidr_ipv6": null,
            "persistence": null,
            "persistence_ttl": null,
            "pool_lb_method": "round-robin",
            "pools": [
                {
                    "name": "issue-01920",
                    "order": null,
                    "ratio": null
                }
            ],
            "provider": {
                "auth_provider": null,
                "no_f5_teem": null,
                "password": null,
                "server": null,
                "server_port": null,
                "timeout": null,
                "transport": "rest",
                "user": null,
                "validate_certs": null
            },
            "state": "present",
            "type": "a",
            "wide_ip": "*.apps.example.com"
        }
    }
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Assert Create wide IP - Idempotent check] *******************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:55
ok: [bigip1] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Remove wide IP] *********************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:61
<10.144.72.9> ESTABLISH LOCAL CONNECTION FOR USER: papineni
<10.144.72.9> EXEC /bin/sh -c 'echo ~papineni && sleep 0'
<10.144.72.9> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/papineni/.ansible/tmp `"&& mkdir "` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076858.69858-36805-221034396255276 `" && echo ansible-tmp-1619076858.69858-36805-221034396255276="` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076858.69858-36805-221034396255276 `" ) && sleep 0'
Using module file /Users/papineni/ansible_repo/f5-ansible/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_gtm_wide_ip.py
<10.144.72.9> PUT /Users/papineni/.ansible/tmp/ansible-local-343978uc5_sjj/tmpdzcq800m TO /Users/papineni/.ansible/tmp/ansible-tmp-1619076858.69858-36805-221034396255276/AnsiballZ_bigip_gtm_wide_ip.py
<10.144.72.9> EXEC /bin/sh -c 'chmod u+x /Users/papineni/.ansible/tmp/ansible-tmp-1619076858.69858-36805-221034396255276/ /Users/papineni/.ansible/tmp/ansible-tmp-1619076858.69858-36805-221034396255276/AnsiballZ_bigip_gtm_wide_ip.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'F5_SERVER=10.144.72.9 F5_USER=admin F5_PASSWORD=F5site02 F5_SERVER_PORT=443 F5_VALIDATE_CERTS=False F5_TEEM=True /Users/papineni/ansible_dev/bin/python /Users/papineni/.ansible/tmp/ansible-tmp-1619076858.69858-36805-221034396255276/AnsiballZ_bigip_gtm_wide_ip.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'rm -f -r /Users/papineni/.ansible/tmp/ansible-tmp-1619076858.69858-36805-221034396255276/ > /dev/null 2>&1 && sleep 0'
changed: [bigip1] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "aliases": null,
            "irules": null,
            "last_resort_pool": null,
            "name": "*.apps.example.com",
            "partition": "Common",
            "persist_cidr_ipv4": null,
            "persist_cidr_ipv6": null,
            "persistence": null,
            "persistence_ttl": null,
            "pool_lb_method": null,
            "pools": null,
            "provider": {
                "auth_provider": null,
                "no_f5_teem": null,
                "password": null,
                "server": null,
                "server_port": null,
                "timeout": null,
                "transport": "rest",
                "user": null,
                "validate_certs": null
            },
            "state": "absent",
            "type": "a",
            "wide_ip": "*.apps.example.com"
        }
    }
}

TASK [bigip_gtm_wide_ip : Issue 01920 - Deprovision GTM] ********************************************************************************************************************************
task path: /Users/papineni/ansible_repo/f5-ansible/test/integration/targets/bigip_gtm_wide_ip/tasks/environment/issue-01920.yaml:69
<10.144.72.9> ESTABLISH LOCAL CONNECTION FOR USER: papineni
<10.144.72.9> EXEC /bin/sh -c 'echo ~papineni && sleep 0'
<10.144.72.9> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/papineni/.ansible/tmp `"&& mkdir "` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076870.485545-36825-99188604850829 `" && echo ansible-tmp-1619076870.485545-36825-99188604850829="` echo /Users/papineni/.ansible/tmp/ansible-tmp-1619076870.485545-36825-99188604850829 `" ) && sleep 0'
Using module file /Users/papineni/ansible_repo/f5-ansible/ansible_collections/f5networks/f5_modules/plugins/modules/bigip_provision.py
<10.144.72.9> PUT /Users/papineni/.ansible/tmp/ansible-local-343978uc5_sjj/tmp1h3l86rk TO /Users/papineni/.ansible/tmp/ansible-tmp-1619076870.485545-36825-99188604850829/AnsiballZ_bigip_provision.py
<10.144.72.9> EXEC /bin/sh -c 'chmod u+x /Users/papineni/.ansible/tmp/ansible-tmp-1619076870.485545-36825-99188604850829/ /Users/papineni/.ansible/tmp/ansible-tmp-1619076870.485545-36825-99188604850829/AnsiballZ_bigip_provision.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'F5_SERVER=10.144.72.9 F5_USER=admin F5_PASSWORD=F5site02 F5_SERVER_PORT=443 F5_VALIDATE_CERTS=False F5_TEEM=True /Users/papineni/ansible_dev/bin/python /Users/papineni/.ansible/tmp/ansible-tmp-1619076870.485545-36825-99188604850829/AnsiballZ_bigip_provision.py && sleep 0'
<10.144.72.9> EXEC /bin/sh -c 'rm -f -r /Users/papineni/.ansible/tmp/ansible-tmp-1619076870.485545-36825-99188604850829/ > /dev/null 2>&1 && sleep 0'
changed: [bigip1] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "level": "nominal",
            "memory": null,
            "module": "gtm",
            "provider": {
                "auth_provider": null,
                "no_f5_teem": null,
                "password": null,
                "server": null,
                "server_port": null,
                "timeout": null,
                "transport": "rest",
                "user": null,
                "validate_certs": null
            },
            "state": "absent"
        }
    }
}
